tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+        unique string used to idenify a Slack channel. Used to send messages to the channel
+    user_id : str
+        the Slack user_id of the user who submitted the request. Used to send ephemeral
+        messages to the user
+    """
+
+    p1 = 0
+    p2 = 0
+    p3 = 0
+    p4 = 0
+    esc_count = 0
+    report = ('This is the current state of Google Cloud Support cases:'
+                '\n\ncase,priority,title,escalated,create_time,last_updated,state,case_creator')
+    with open(cases_file) as f:
+        cases = json.load(f)
+    
+    for case in cases:
+        if cases[case]['priority'] == 'P1':
+            p1 += 1
+        elif cases[case]['priority'] == 'P2':
+            p2 += 1
+        elif cases[case]['priority'] == 'P3':
+            p3 += 1
+        else:
+            p4 += 1
+
+        if cases[case]['escalated'] == True:
+            esc_count += 1
+        
+        report = report + '\n{},{},{},{},{},{},{},{}'.format(
+            case, 
+            cases[case]['priority'], 
+            cases[case]['case_title'],
+            cases[case]['escalated'],
+            cases[case]['create_time'], 
+            cases[case]['update_time'],
+            cases[case]['state'],
+            cases[case]['case_creator'])
+
+    report = (report + '\n\n'
+        '\n{} P1 cases are open'
+        '\n{} P2 cases are open'
+        '\n{} P3 cases are open'
+        '\n{} P4 cases are open'
+        '\nTotal cases open: {}'
+        '\nEscalated cases: {}'
+        ).format(str(p1), str(p2), str(p3), str(p4), str(p1 + p2 + p3 + p4), str(esc_count))
+
+    client.chat_postMessage(channel=channel_id, text=f"{report}")
+
+
+
+def case_updates():
+    """
+    Infinite loop that pulls all of the open Google Cloud support cases for our org and their 
+    associated public comments every 15 seconds and compares it to the cases and 
+    comments from the previous pull. If any change is detected between the two versions 
+    of the case, the change is posted to any channel that is tracking it
+    """
+    
+    query_string = 'organization="organizations/{}" AND state=OPEN'.format(ORG_ID)
+
+    while True:
+        loop_skip = False
+        sleep_timer = 15
+        cases = {}
+        try:
+            if os.path.exists(cases_file):
+                with open(cases_file) as f:
+                    cases = json.load(f)
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            pass
+
+        req = support_service.cases().search(query=query_string)
+        try:
+            resp = req.execute().get('cases', [])
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            time.sleep(5)
+            continue
+
+        temp_cases = {}
+        
+        for case in resp:
+            try:
+                temp_case = Support_Case(case)
+                temp_cases[temp_case.case_number] = temp_case
+            except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.